packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -446,39 +452,59 @@ void main() {
         ),
       ),
     ));
-    expect(find.text('bar1'), findsNothing);
-    expect(find.text('bar2'), findsNothing);
-    await tester.tap(find.byKey(tapTarget)); // queue bar1
-    await tester.tap(find.byKey(tapTarget)); // queue bar2
-    expect(find.text('bar1'), findsNothing);
-    expect(find.text('bar2'), findsNothing);
-    await tester.pump(); // schedule animation for bar1
-    expect(find.text('bar1'), findsOneWidget);
-    expect(find.text('bar2'), findsNothing);
-    await tester.pump(); // begin animation
-    expect(find.text('bar1'), findsOneWidget);
-    expect(find.text('bar2'), findsNothing);
-    await tester.pump(const Duration(milliseconds: 750)); // 0.75s // animation last frame; two second timer starts here
-    await tester.drag(find.text('bar1'), const Offset(0.0, 50.0));
-    await tester.pump(); // bar1 dismissed, bar2 begins animating
-    expect(find.text('bar1'), findsNothing);
-    expect(find.text('bar2'), findsOneWidget);
+
+    final Map<DismissDirection, List<Offset>> dragGestures = _getDragGesturesOfDismissDirections(width);
+
+    for (dismissDirection in dragGestures.keys) {
+      for (final Offset dragGesture in dragGestures[dismissDirection]!) {
+        snackBarCount = 0;
+
+        expect(find.text('bar1'), findsNothing);
+        expect(find.text('bar2'), findsNothing);
+        await tester.tap(find.byKey(tapTarget)); // queue bar1
+        await tester.tap(find.byKey(tapTarget)); // queue bar2
+        expect(find.text('bar1'), findsNothing);
+        expect(find.text('bar2'), findsNothing);
+        await tester.pump(); // schedule animation for bar1
+        expect(find.text('bar1'), findsOneWidget);
+        expect(find.text('bar2'), findsNothing);
+        await tester.pump(); // begin animation
+        expect(find.text('bar1'), findsOneWidget);
+        expect(find.text('bar2'), findsNothing);
+        await tester.pump(const Duration(milliseconds: 750)); // 0.75s // animation last frame; two second timer starts here
+        await tester.drag(find.text('bar1'), dragGesture);
+        await tester.pump(); // bar1 dismissed, bar2 begins animating
+        expect(find.text('bar1'), findsNothing);
+        expect(find.text('bar2'), findsOneWidget);
+        await tester.pump(const Duration(milliseconds: 750)); // 0.75s // animation last frame; two second timer starts here
+        await tester.drag(find.text('bar2'), dragGesture);
+        await tester.pump(); // bar2 dismissed
+        expect(find.text('bar1'), findsNothing);
+        expect(find.text('bar2'), findsNothing);
+      }
+    }
```

Comment:
This section of code 456-485 looks like it can be consolidated as well, it is the same as 521-550

Reply:
Committed: 2635f58